From db45037c75851c809e0a6890f65b3abd827a36c8 Mon Sep 17 00:00:00 2001
From: Vinod Koul <vinod.koul@intel.com>
Date: Wed, 26 Aug 2015 13:13:56 +0530
Subject: [PATCH 0211/1214] ASoC: Intel: Skylake: Add debugfs NHLT dmic
 override

Add debugfs entries for reading and writing DMIC blobs which
driver can use to program DSP

Signed-off-by: Vinod Koul <vinod.koul@intel.com>
---
 sound/soc/intel/skylake/skl-debug.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sound/soc/intel/skylake/skl-debug.c b/sound/soc/intel/skylake/skl-debug.c
index 9f3a596..57d5a79 100644
--- a/sound/soc/intel/skylake/skl-debug.c
+++ b/sound/soc/intel/skylake/skl-debug.c
@@ -43,6 +43,7 @@ struct skl_debug {
 	struct dentry *nhlt;
 	u8 fw_read_buff[FW_REG_BUF];
 	struct nhlt_blob ssp_blob[MAX_SSP];
+	struct nhlt_blob dmic_blob;
 };
 
 static ssize_t nhlt_read(struct file *file, char __user *user_buf,
@@ -360,6 +361,11 @@ static int skl_init_nhlt(struct skl_debug *d)
 			dev_err(d->dev, "%s: debugfs init failed\n", name);
 	}
 
+	if (!debugfs_create_file("dmic", 0644,
+				d->nhlt, &d->dmic_blob,
+				&nhlt_fops))
+		dev_err(d->dev, "%s: debugfs init failed\n", name);
+
 	return 0;
 }
 
-- 
2.7.4

